### PR TITLE
Update README.md for custom domain

### PR DESCRIPTION
**Pull request by dlepow**
Note key must be exportable

### CODE REVIEW

File: docs/custom-domain/README.md [hunk 1]
```diff
@@ -1,105 +1,125 @@
...
 - Custom registry domain details
-  - custom registry domain (`container-registry.contoso.com`)
-  - key vault secret ID of the corresponding TLS data
-  - client ID of the user assigned registry identity that has access to this secret (not required in case of system assigned)
-- Custom data domain details
-  - regional custom data domain (`eastus-registry-data.contoso.com`)
-  - key vault secret ID of the corresponding TLS data
-  - client ID of the user assigned registry identity that has access to this secret (not required in case of system assigned)
+  - custom registry domain name (`container-registry.contoso.com`)
+  - key vault certificate ID of the corresponding TLS data (a URI of the form `https://myvaultvault.azure.net/certificates/mycertificate/xxxxxxxxxxxxx`)
```
Comment (avtakkar) [inline docs/custom-domain/README.md:120]: This needs to remain a secret ID. Certificate IDs do not have the private key information.

Comment (dlepow) [reply docs/custom-domain/README.md]: Fixed. Thanks for clarifying

**Comment by dlepow** [comment]
@toddysm @luisdlp @SteveLasker @yugangw-msft @avtakkar - Updated the custom domain article with "private preview" disclaimers, polished the text a little, added a few clarifications around where to find IDs, etc. Please have a look when you have a chance. Thank you!

File: docs/custom-domain/README.md [hunk 2]
```diff
@@ -1,105 +1,128 @@
...
+This article describes the steps to set up the endpoints of an Azure container registry using your own custom domain name, such as **contoso.com**.
 
-The following steps describe how you can achieve this.
+By default, every Azure container registry is accessed externally using its login server name. If you have a registry called `myregistry`, you access it using its default hostname, `myregistry.azurecr.io` (in the Azure Cloud). As a customer belonging to an organization, you may prefer to access your registry using a custom domain name that is associated with your organization, for example, `myregistry.contoso.com`.
+
+**Important - Using a custom domain name in Azure Container Registry is currently a private preview feature**. It can only be enabled by preparing certificates, a Premium container registry, and an Azure key vault as described in this article, and then opening an Azure support request to complete the configuration. 
+
+## Limitations
+
+* This capability is provided without a service level agreement, and isn't currently recommended for production workloads.
```
Comment (SteveLasker) [inline docs/custom-domain/README.md:11]: @sajayantony
Are we not willing to support customers with custom domains?
Or, are we only willing to support a subset of customers, so not all requests will be answered? Just seems the only customers that would want custom domains would be customers in production environments that need the extra security.

File: docs/custom-domain/README.md [hunk 3]
```diff
@@ -1,105 +1,128 @@
...
+* Azure Container Registry custom domains can't currently be used where key vault access is restricted.
```
Comment (SteveLasker) [inline docs/custom-domain/README.md:12]: Might want to elaborate on what "restricted means". Are we talking about auth, or private link?

File: docs/custom-domain/README.md [hunk 4]
```diff
@@ -1,105 +1,128 @@
...
+- [Azure CLI](https://docs.microsoft.com/cli/azure/): version 2.4.0 or higher, or use [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview)
+- An Azure container registry in the Premium service tier. See [this article](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-azure-cli) for instructions to create one. If you have an existing registry, you can [upgrade](https://docs.microsoft.com/azure/container-registry/container-registry-skus#changing-tiers).
 - Your custom domain names. The following two are required:
   - Custom registry domain to access the registry REST API. Example for the `contoso.com` domain: `container-registry.contoso.com` 
-  - Custom data domain to access the registry content. Again, example for `contoso.com`: `eastus-registry-data.contoso.com`
-    - Note that the custom data domain is region specific. For geo-replicated registries, each region should have its own custom data endpoint.
+  - Custom data domain to access the registry content. Example for `contoso.com`: `eastus-registry-data.contoso.com`
```
Comment (SteveLasker) [inline docs/custom-domain/README.md:19]: can we follow the same recommendation we've made for ACR:

- REST endpoint: registry.domain (registry.contoso.com)
- Data endpoint: [region].data.registry.contoso.com (eastus.data.registry.contoso.com, westus.data.registry.contoso.com)

**Review by SteveLasker — COMMENTED** [review]
I didn't have a chance to do the full walkthrough. I'll try later, but wanted to provide feedback thus far on support and the domain names.
Thanks for giving this the @dlepow polish

File: docs/custom-domain/README.md (hunk 3 above)
Comment (dlepow) [reply docs/custom-domain/README.md]: Updated to "Azure Container Registry custom domains can't currently be configured using an Azure key vault deployed in a virtual network with a Key Vault firewall."

File: docs/custom-domain/README.md (hunk 4 above)
Comment (dlepow) [reply docs/custom-domain/README.md]: Good suggestion, updated.

**Review by dlepow — COMMENTED** [review]
Thanks @SteveLasker - incorp'd your suggestions. Steve and @sajayantony I'll update the "Limitations" for custom domain based on your input. Perhaps we differentiate what is actually private preview: is it the configuration experience, or is it the use of the custom domain? Thanks

File: docs/custom-domain/README.md (hunk 3 above)
Comment (aviral26) [reply docs/custom-domain/README.md]: If a system assigned managed identity is used, key vault firewall can be bypassed if Trusted Service Bypass is enabled. This is not true of user-assigned managed identities though.

Comment (dlepow) [reply docs/custom-domain/README.md]: @aviral26 - Removed the limitation and added information to "Enhanced security with virtual network"

**Comment by dlepow** [comment]
@SteveLasker @toddysm - Clarified feature status and left remainder of doc as-is. Please merge when you have a chance. Thank you!

**Review by sajayantony — APPROVED** [review]

**Pull request merged**